section-5/exercises/vsnitsar-section5.js

```diff
@@ -0,0 +1,201 @@
+        return "User didn't pass the exam";
+    }
+}
+
+console.log(examResult(95));
+console.log(examResult(82));
+console.log(examResult(75));
+console.log(examResult(65));
+console.log(examResult(50));
+
+/*Exercise 5:
+What will be the result of the code bellow? Think without code execution
+*/
+
+const username = "";
+if (username) {
+    console.log("User exists");
+} else {
+    console.log("No user found");
+}
+
+// empty string in JS it's falsy value so will be returned "No user found"
+
+/*Exercise 6:
+Rewrite the following if-else statement using a ternary operator:
+*/
+
+const temp = 30;
+if (temp > 25) {
+    console.log("Hot day");
+} else {
+    console.log("Cool day");
+}
+
+const isHotDay = temp > 25 ? "Hot day" : "Cool day";
+console.log(isHotDay);
+
+/*Exercise 7:
+What will be the result of the code bellow? Think without code execution:
+*/
+
+const isAdult = false;
+if (!isAdult) {
+    console.log('Truthy')
+} else {
+    console.log('Falsy')
+}
+
+// result will be 'Truthy', because ! of false it's true and true condition will be executed
+
+/*Exercise 8:
+1. What will be the result of the code bellow? Think without code execution:
+*/
+
+const points = 0;
+console.log(points || 100);
+
+// points = 0 it's falsy value, then || check points value if it's true - in this case its not - then will be used
+// second value 100 - so result 100
+
+/*2. What will be the result of the code bellow? Think without code execution:*/
+
+const username = "";
+console.log(username && "Guest");
+
+// in console log will be empty string, because && returns first falsy value - in this example it's username
+
+/*3.What will be the result of the code bellow? Think without code execution:*/
+
+const userCountry = null;
+console.log(userCountry ?? 'USA');
+
+// output 'USA' because user country null, and ?? provide value in case if null or undefined was recognized
+
+/*Exercise 9:
+What will be the result of the code bellow? Think without code execution:
+*/
+
+function getUserData(userData) {
+    return {
+        name: userData.name || 'Default value',
+        year: userData.year || 2005,
+    }
+}
+
+console.log(getUserData({name: 'Test user'})); // test user + 2005 -  because name not falsy + falsy year
+console.log(getUserData({year: 2005})); // default value + 2005 - name falsy + year not falsy but still 2005
+console.log(getUserData({})); // default value + 2005 - both value falsy and used defaults
+
+/*Exercise 10:
+Use all three logical assignment operators (||=, &&=, ??=) in one function:
+*/
+
+function updateUser(userData) {
+    return {
+        name: userData.name ||= 'Guest',
+        age: userData.age &&= 25,
+        premium: userData.premium ??= false,
+    }
+}
+
+const userOne = {name: "", age: null, premium: true};
+const userTwo = {name: "Alice", age: 25, premium: false};
+
+updateUser(userOne);
+updateUser(userTwo);
+
+console.log(userOne);
+
+/** Expected Output:
+ { name: "Guest", age: 18, premium: true }
```

Comment:
Could you review your updateUser function once again and think about result of age property?